src/frameworks/CoreServices/src/CarbonCore/Multiprocessing.cpp

```diff
@@ -109,3 +109,23 @@ OSStatus MPExitCriticalRegion(MPCriticalRegionID criticalRegion)
 	else
 		return paramErr;
 }
+
+OSStatus MPCreateSemaphore(MPSemaphoreCount maximumValue, MPSemaphoreCount initialValue, MPSemaphoreID *semaphore)
+{
+    return NULL;
+}
+
+OSStatus MPDeleteSemaphore(MPSemaphoreID semaphore)
+{
+    return NULL;
+}
+
+OSStatus MPSignalSemaphore(MPSemaphoreID semaphore)
+{
+    return NULL;
+}
+
+OSStatus MPWaitOnSemaphore(MPSemaphoreID semaphore, Duration timeout)
+{
+    return NULL;
+}
```

Comment:
Might be a good idea to add logging to the stubs

Include the following near the top of the source file.
```c
static int verbose = 0;
__attribute__((constructor)) static void initme(void) {
    verbose = getenv("STUB_VERBOSE") != NULL;
}
```

Make the following changes to the methods.
```suggestion
OSStatus MPCreateSemaphore(MPSemaphoreCount maximumValue, MPSemaphoreCount initialValue, MPSemaphoreID *semaphore)
{
    if (verbose) puts("STUB: MPCreateSemaphore called");
    return NULL;
}

OSStatus MPDeleteSemaphore(MPSemaphoreID semaphore)
{
    if (verbose) puts("STUB: MPDeleteSemaphore called");
    return NULL;
}

OSStatus MPSignalSemaphore(MPSemaphoreID semaphore)
{
    if (verbose) puts("STUB: MPSignalSemaphore called");
    return NULL;
}

OSStatus MPWaitOnSemaphore(MPSemaphoreID semaphore, Duration timeout)
{
    if (verbose) puts("STUB: MPWaitOnSemaphore called");
    return NULL;
}
```

---

`OSStatus` is a 32bit (signed) integer. `NULL` is normally used for pointers. 
```c
// Look in `MacTypes.h`
#if __LP64__
typedef signed int                      SInt32;
#else
typedef signed long                     SInt32;
#endif
typedef SInt32                          OSStatus;
```

I would replace `NULL` with `noErr`
```suggestion
OSStatus MPCreateSemaphore(MPSemaphoreCount maximumValue, MPSemaphoreCount initialValue, MPSemaphoreID *semaphore)
{
    return noErr;
}

OSStatus MPDeleteSemaphore(MPSemaphoreID semaphore)
{
    return noErr;
}

OSStatus MPSignalSemaphore(MPSemaphoreID semaphore)
{
    return noErr;
}

OSStatus MPWaitOnSemaphore(MPSemaphoreID semaphore, Duration timeout)
{
    return noErr;
}
```